curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/inpaint.html

```diff
@@ -0,0 +1,129 @@
+<html>
+<head>
+<meta charset="utf-8">
+<title>CVdrawing</title>
+</head>
+<body>
+<p id="status">OpenCV.js is loading...</p>
+<canvas id="canvas" width="640" height="426">
+</canvas>
+<img id="src" src="bird.jpg" style="display:none"/>
+<input type="button" id="myButton" value="Test" disabled=true
+onclick="inpaint()"/>
+
+<script type="text/javascript">
+
+var checker = false;
+var src;
+var mask;
+var dst;
+var p_x = 0;
+var p_y = 0;
+var paint;
+var srcmat;
+var maskmat;
+var dstmat;
+var x;
+var y;
+
+window.onload = function()
+{
+context = document.getElementById("canvas").getContext("2d");
+img = document.getElementById("src");
+context.drawImage(img, 0, 0);
+};
+
+document.getElementById("canvas").onmousedown = function(e)
+{
+  paint = true;
+  x = e.pageX - this.offsetLeft;
+  y = e.pageY - this.offsetTop;
+  if (p_x == 0)
+  {
+      p_x = x;
+      p_y = y;
+  }
+  draw();
+};
+
+document.getElementById("canvas").onmousemove = function(e)
+{
+  if(paint)
+  {
+    cv.line(src, new cv.Point(p_x, p_y), new cv.Point(x, y), new cv.Scalar(255, 255, 255, 255), 2);
+    cv.line(maskmat, new cv.Point(p_x, p_y), new cv.Point(x, y), new cv.Scalar(255, 255, 255, 255), 2);
+    p_x = x;
+    p_y = y;
+    x = e.pageX - this.offsetLeft;
+    y = e.pageY - this.offsetTop;
+    draw();
+  }
+};
+
+document.getElementById("canvas").onmouseup = function(e)
+{
+  paint = false;
+  p_x = 0;
+  p_y = 0;
+};
```

Comment:
Can we remove `paint` variable and detect it by `p_x` and `p_y` values? In example, they can be `undefined` instead of 0 as an indicator that drawing is not started.

Reply:
In JS it is like "null" ?

Reply:
I tested few cases for this remark:
1) If we "null" them, we have type error, because we can not convert "null" to int;
2) If we not defined them, we have first line started from corner (0, 0).
As consequences, we should use "paint" as Boolean variable, for checking drawing start and correct drawing.

Reply:
@ilbalance, Can you please specify the line with this error?

Reply:
I tried to change lastX and lastY in strings #19 and #20.

Reply:
Boolean variable "paint" replaced with "if(lastX != undefined)" statements.
Sample tested, works fine.

Reply:
Nice, thank you!